sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFn.java

```diff
@@ -614,45 +612,93 @@ public Duration getAllowedTimestampSkew() {
    * <p>See <a href="https://s.apache.org/splittable-do-fn">the proposal</a> for an overview of the
    * involved concepts (<i>splittable DoFn</i>, <i>restriction</i>, <i>restriction tracker</i>).
    *
-   * <p>If a {@link DoFn} is splittable, the following constraints must be respected:
+   * <p>A splittable {@link DoFn} must obey the following constraints:
    *
    * <ul>
    *   <li>It <i>must</i> define a {@link GetInitialRestriction} method.
-   *   <li>It <i>may</i> define a {@link GetSize} method.
-   *   <li>It <i>may</i> define a {@link SplitRestriction} method.
+   *   <li>It <i>should</i> define a {@link GetSize} method or ensure that the {@link
+   *       RestrictionTracker} implements {@link Sizes.HasSize}.
+   *   <li>It <i>should</i> define a {@link SplitRestriction} method.
    *   <li>It <i>may</i> define a {@link NewTracker} method returning a subtype of {@code
    *       RestrictionTracker<R>} where {@code R} is the restriction type returned by {@link
    *       GetInitialRestriction}. This method is optional in case the restriction type returned by
    *       {@link GetInitialRestriction} implements {@link HasDefaultTracker}.
    *   <li>It <i>may</i> define a {@link GetRestrictionCoder} method.
    *   <li>The type of restrictions used by all of these methods must be the same.
-   *   <li>Its {@link ProcessElement} method <i>may</i> return a {@link ProcessContinuation} to
-   *       indicate whether there is more work to be done for the current element.
-   *   <li>Its {@link ProcessElement} method <i>must not</i> use any extra context parameters, such
-   *       as {@link BoundedWindow}.
    *   <li>The {@link DoFn} itself <i>may</i> be annotated with {@link BoundedPerElement} or {@link
    *       UnboundedPerElement}, but not both at the same time. If it's not annotated with either of
    *       these, it's assumed to be {@link BoundedPerElement} if its {@link ProcessElement} method
    *       returns {@code void} and {@link UnboundedPerElement} if it returns a {@link
    *       ProcessContinuation}.
+   *   <li>Timers and state must not be used.
    * </ul>
    *
    * <p>A non-splittable {@link DoFn} <i>must not</i> define any of these methods.
+   *
+   * <p>This method must satisfy the following constraints:
```

Comment:
Would be helpful to add an "If this DoFn is splittable," here just to be very clear.

Reply:
Done